actions/user_actions.jsx

```diff
@@ -89,78 +102,41 @@ export async function loadProfilesWithoutTeam(page, perPage, success) {
-            continue;
-        }
-
-        const members = getChannelMembersInChannels(getState())[channelId];
-        if (!members || !members[pid]) {
-            membersToLoad[pid] = true;
-        }
-    }
-
-    const list = Object.keys(membersToLoad);
-    if (list.length === 0) {
-        if (success) {
-            success({});
-        }
-        return;
-    }
-
-    dispatch(getChannelMembersByIds(channelId, list)).then((result) => {
-        if (result.error) {
-            error(result.error);
-        } else {
-            success(result.data);
-        }
-    });
-}
+export function loadTeamMembersAndChannelMembersForProfilesList(profiles, teamId, channelId) {
+    return async (doDispatch, doGetState) => {
+        const state = doGetState();
+        const teamIdParam = teamId || getCurrentTeamId(state);
+        const channelIdParam = channelId || getCurrentChannelId(state);
+        doDispatch(loadTeamMembersForProfilesList(profiles, teamIdParam)).then(() => {
+            doDispatch(loadChannelMembersForProfilesList(profiles, channelIdParam));
+        });
 
-export function loadTeamMembersAndChannelMembersForProfilesList(profiles, teamId = getCurrentTeamId(getState()), channelId = getCurrentChannelId(getState()), success, error) {
-    loadTeamMembersForProfilesList(profiles, teamId, () => {
-        loadChannelMembersForProfilesList(profiles, channelId, success, error);
-    }, error);
+        return {data: true};
+    };
 }
 
-export function loadChannelMembersForProfilesList(profiles, channelId = getCurrentChannelId(getState()), success, error) {
-    const membersToLoad = {};
-    for (let i = 0; i < profiles.length; i++) {
-        const pid = profiles[i].id;
+export function loadChannelMembersForProfilesList(profiles, channelId) {
+    return (doDispatch, doGetState) => {
+        const state = doGetState();
+        const channelIdParam = channelId || getCurrentChannelId(state);
+        const membersToLoad = {};
+        for (let i = 0; i < profiles.length; i++) {
+            const pid = profiles[i].id;
 
-        const members = getChannelMembersInChannels(getState())[channelId];
-        if (!members || !members[pid]) {
-            membersToLoad[pid] = true;
+            const members = getChannelMembersInChannels(state)[channelIdParam];
+            if (!members || !members[pid]) {
+                membersToLoad[pid] = true;
+            }
         }
-    }
 
-    const list = Object.keys(membersToLoad);
-    if (list.length === 0) {
-        if (success) {
-            success({});
+        const list = Object.keys(membersToLoad);
+        if (list.length === 0) {
+            return {data: true};
         }
-        return;
-    }
 
-    dispatch(getChannelMembersByIds(channelId, list)).then((result) => {
-        if (result.error) {
-            error(result.error);
-        } else {
-            success(result.data);
-        }
-    });
+        doDispatch(getChannelMembersByIds(channelIdParam, list));
```

Comment:
If we don't await here, we are returning on `.then` before the `getChannelMembersById` action is run.